### DISCUSSION TITLE
How to set GWL in 3D model?

### DISCUSSION THREAD

**Discussion by Aping1100** [Anura3D/Anura3D_OpenSource#65] [Q&A] [answered]
Hi everyone,
I am trying to build a 3D slope model with groundwater. However, I have no idea how to set initial condition by building PhreaticSurface file (TXT) like the way in building 2D models.
Hope for your answers.

**Comment by FrancescaCeccato** [comment]
We haven't tested the 3D option yet. if you are willing to implement it please let us know and we'll help.

**Reply by Aping1100** [reply]
I truly appreciate your support.
I have tested the phreatic surface by applying to a cubic model. It seems that the phreatic surface can only follow the setting in x-axis, and it will directly project to the remaining axes as shown in the figure below. Since I found that it is common to see a thin impermeable layer between two formations, which make the soil layers with different groundwater levels.  Is it possible to set different phreatic surfaces for different soil layers on initial condition?
![3d_gwl_test](https://user-images.githubusercontent.com/104591392/222056315-0fa3ec2b-a032-4e57-b1af-ea69c2431b5b.PNG)

**Comment by gdicarluccio** [comment, marked as answer]
Yes, it is possible to apply different initial water levels to different materials. 
In the pre-processing you can assign to surfaces/volumes, occupied by different materials, different text files including the water level profile.

**Reply by Aping1100** [reply]
Thanks for the reply.
I have tried this method before, but the results only show that two soil layer with the same phreatic surface. In addition, in GOM file there is only one piece of water surface data. Did I miss any step?
![2GWL](https://user-images.githubusercontent.com/104591392/222340757-f0925f76-b0b8-4f14-82e1-b4b95eefe330.PNG)

**Reply by gdicarluccio** [reply]
To the two surfaces, red and green, are assigned different materials?
Did you try to remesh the model?

**Reply by Aping1100** [reply]
It works!
Thanks for your help. 
I used the same material for both soil layers, which resulted in failure.
![擷取](https://user-images.githubusercontent.com/104591392/222641389-c04b32c3-6771-4f30-8088-98a22a07e209.PNG)
